src/server/unit_sao.cpp

```diff
@@ -38,6 +38,91 @@ ServerActiveObject *UnitSAO::getParent() const
 	return obj;
 }
 
+void UnitSAO::setRotation(v3f rotation)
+{
+	ObjectProperties *props = accessObjectProperties();
+
+	// If any of those two properties is set as 'true', the corresponding box
+	// will rotate automatically on each rotation change of the entity
+
+	if (props->synchronize_cbox_rotation_with_dir) {
+		infostream << "rotating collisionbox" << std::endl;
+		setBoxRotation(rotation, "collision");
+	}
+
+	if (props->synchronize_sbox_rotation_with_dir) {
+		infostream << "rotating selectionbox" << std::endl;
+		setBoxRotation(rotation, "selection");
+	}
+
+	m_rotation = rotation;
+}
+
+void UnitSAO::setBoxRotation(v3f rotation, const std::string &box_type)
+{
+	ObjectProperties *props = accessObjectProperties();
+
+	aabb3f *box = box_type == "collision"
+					? &props->collisionbox
+					: (box_type == "selection" ? &props->selectionbox 
+					   				: nullptr);
```

Comment:
How many tabs is it necessary to do here?

Reply:
I think you pull in the first two line line by one tab, and third by two tabs.

But perhaps this is nicer as an if-statement?
```
aabb3f *box = nullptr;
if (box_type == "collision)
        box = &props->collisionbox;
else if (box_type == "selection")
        box = &props->selectionbox;
```

But either is fine.